lib/player.js

```diff
@@ -881,6 +884,18 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
 
     await detachPromise;
 
+    // A PreloadManager can only be used with the Player instance that created
+    // it, so all PreloadManagers this Player has created are now useless.
+    // Destroy any remaining managers now, to help prevent memory leaks.
+    const preloadManagerDestroys = [];
+    for (const preloadManager of this.createdPreloadManagers_) {
+      if (!preloadManager.isDestroyed()) {
+        preloadManagerDestroys.push(preloadManager.destroy());
+      }
+    }
+    await Promise.all(preloadManagerDestroys);
+    this.createdPreloadManagers_ = [];
```

Comment:
I would have preferred this above `await`.  Probably not crucial, but ideally you would empty that array right after initiating all those promises, rather than after they complete.